File: app/src/main/java/pro/butovanton/mega/ViewModelDetail.java
package pro.butovanton.mega;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;


class ViewModelDetail extends ViewModel {

    private NetworkService networkService;
    private JSONPlaceHolderApi jsonPlaceHolderApi;
    MutableLiveData<ModelDetail> modelDetail = new MutableLiveData<ModelDetail>();

    public ViewModelDetail() {
        networkService = NetworkService.getInstance();
        jsonPlaceHolderApi = networkService.getJSONApi();
    }

    public LiveData<ModelDetail> getModelDetail(int id)    {

        jsonPlaceHolderApi.getModelDetail(id).enqueue(new Callback<List<ModelDetail>>() {
            @Override
            public void onResponse(Call<List<ModelDetail>> call, Response<List<ModelDetail>> response) {
                ModelDetail modelDet = response.body().get(0);
                modelDet.setImg(NetworkService.BASE_URL + modelDet.getImg());
                modelDetail.setValue(modelDet);
            }

            @Override
            public void onFailure(Call<List<ModelDetail>> call, Throwable t) {
                Log.d("DEBUG", t.toString());
            }
        });
        return modelDetail ;
    }

}

File: app/src/main/java/pro/butovanton/mega/ViewModelDetail.kt
package pro.butovanton.mega

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import pro.butovanton.mega.NetworkService.Companion.instance
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

internal class ViewModelDetail : ViewModel() {
    private val networkService: NetworkService?
    private val jsonPlaceHolderApi: JSONPlaceHolderApi
    var modelDetail = MutableLiveData<ModelDetail>()
    fun getModelDetail(id: Int): LiveData<ModelDetail> {
        val call = jsonPlaceHolderApi.getModelDetail(id) as Call<List<ModelDetail>>
        call.enqueue(object : Callback<List<ModelDetail>> {
            override fun onResponse(call: Call<List<ModelDetail>>, response: Response<List<ModelDetail>>) {
                val modelDet = response.body()!![0]
                modelDet.img = NetworkService.BASE_URL + modelDet.img
                modelDetail.setValue(modelDet)
            }

            override fun onFailure(call: Call<List<ModelDetail>>, t: Throwable) {
                Log.d("DEBUG", t.toString())
            }
        })
        return modelDetail
    }

    init {
        networkService = instance
        jsonPlaceHolderApi = networkService!!.jSONApi
    }
}

